homeassistant/components/weatherkit/config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Adds config flow for WeatherKit."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from apple_weatherkit.client import (
+    WeatherKitApiClient,
+    WeatherKitApiClientAuthenticationError,
+    WeatherKitApiClientCommunicationError,
+    WeatherKitApiClientError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    LocationSelector,
+    LocationSelectorConfig,
+    TextSelector,
+    TextSelectorConfig,
+)
+
+from .const import (
+    CONF_KEY_ID,
+    CONF_KEY_PEM,
+    CONF_SERVICE_ID,
+    CONF_TEAM_ID,
+    DOMAIN,
+    LOGGER,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Home"): str,
+        vol.Required(CONF_LOCATION): LocationSelector(
+            LocationSelectorConfig(radius=False, icon="")
+        ),
+        # Auth
+        vol.Required(CONF_KEY_ID): str,
+        vol.Required(CONF_SERVICE_ID): str,
+        vol.Required(CONF_TEAM_ID): str,
+        vol.Required(CONF_KEY_PEM): TextSelector(
+            TextSelectorConfig(
+                multiline=True,
+            )
+        ),
+    }
+)
+
+
+class WeatherKitUnsupportedLocationError(Exception):
+    """Error to indicate a location is unsupported."""
+
+
+class WeatherKitFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for WeatherKit."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self,
+        user_input: dict | None = None,
+    ) -> data_entry_flow.FlowResult:
+        """Handle a flow initialized by the user."""
+        _errors = {}
```

Comment:
```suggestion
        errors = {}
```
Doesn't really need to be private imo